source/CardGames.py

```diff
@@ -191,12 +191,34 @@ def resetDeck(self):
     self.deck.reset()
     self.deck.shuffle()
 
-
+  
 class Poker:
   def __init__(self, players: "list[Player]", dealer: Dealer):
     self.players = players
     self.dealer = dealer
     self.pot = 0
-
+  
   # Add all functions pertaining to the game of poker below
   # Add global variables to the __init__ function above
+  
+  def check_flush(self, hand: "list[Card]") -> bool:
+    # Check if the given hand is a flush
+    suits_count = {'Spades': 0, 'Clubs': 0, 'Hearts': 0, 'Diamonds': 0}
+    for card in hand:
+        suits_count[card.suit] += 1
+
+    # Check if any suit has 5 or more cards
+    for count in suits_count.values():
+        if count >= 5:
+            return True
+    return False
+
+  
+  def evaluate_hands(self):
+      for player in self.players:
+        if self.check_flush(player.hand):
+          print(f"{player.name} has a flush!")
+        else:
+          print(f"{player.name} does not have a flush.")
+
```

Comment:
I don't see any logic errors in your def check_flush() as it is updating and checking if a hand has all 5 cards in the same suit through the use of values() and def evaluate_hands()! Good job!

Reply:
Thanks Erin!